code/datums/components/crafting/cooking.dm

```diff
@@ -0,0 +1,492 @@
+			If there isn't enough take all of that reagent from the container, put into deletion list, substract the amt var by the volume of reagent, remove the container from surroundings list and keep searching
+			While doing above stuff check deletion list if it already has such reagnet, if yes merge instead of adding second one
+		If its stack check if it has enough amount
+			If yes create new stack with the needed amount and put in into deletion list, substract taken amount from the stack
+			If no put all of the stack in the deletion list, substract its amount from amt and keep searching
+			While doing above stuff check deletion list if it already has such stack type, if yes try to merge them instead of adding new one
+		If its anything else just locate() in in the list in a while loop, each find --s the amt var and puts the found stuff in deletion loop
+
+	Then do a loop over parts var of the recipe
+		Do similar stuff to what we have done above, but now in deletion list, until the parts conditions are satisfied keep taking from the deletion list and putting it into parts list for return
+
+	After its done loop over deletion list and delete all the shit that wasn't taken by parts loop
+
+	del_reqs return the list of parts resulting object will receive as argument of CheckParts proc, on the atom level it will add them all to the contents, on all other levels it calls ..() and does whatever is needed afterwards but from contents list already
+*/
+
+/datum/component/cooking/proc/del_reqs(datum/crafting_recipe/R, atom/a)
+	var/list/surroundings
+	var/list/Deletion = list()
+	. = list()
+	var/data
+	var/amt
+	var/list/requirements = list()
+	if(R.reqs)
+		requirements += R.reqs
+	if(R.machinery)
+		requirements += R.machinery
+	main_loop:
+		for(var/path_key in requirements)
+			amt = R.reqs[path_key] || R.machinery[path_key]
+			if(!amt)//since machinery can have 0 aka CRAFTING_MACHINERY_USE - i.e. use it, don't consume it!
+				continue main_loop
+			surroundings = get_environment(a, R.blacklist)
+			surroundings -= Deletion
+			if(ispath(path_key, /datum/reagent))
+				var/datum/reagent/RG = new path_key
+				var/datum/reagent/RGNT
+				while(amt > 0)
+					var/obj/item/reagent_containers/RC = locate() in surroundings
+					RG = RC.reagents.get_reagent(path_key)
+					if(RG)
+						if(!locate(RG.type) in Deletion)
+							Deletion += new RG.type()
+						if(RG.volume > amt)
+							RG.volume -= amt
+							data = RG.data
+							RC.reagents.conditional_update(RC)
+							RC.update_appearance(UPDATE_ICON)
+							RG = locate(RG.type) in Deletion
+							RG.volume = amt
+							RG.data += data
+							continue main_loop
+						else
+							surroundings -= RC
+							amt -= RG.volume
+							RC.reagents.reagent_list -= RG
+							RC.reagents.conditional_update(RC)
+							RC.update_appearance(UPDATE_ICON)
+							RGNT = locate(RG.type) in Deletion
+							RGNT.volume += RG.volume
+							RGNT.data += RG.data
```

Comment:
Even though this isn't a one-letter variable name, this feels like violating our contribution rules. Use proper names, like `reagent`, `client`, `cook`, etc.

Reply:
Most of the backend, including this code is copied from the crafting menu, I will leave it as is for now because I want to use the code diff later (when I move the crafting part to the new UI), remove the original file and then refactor the new one.